plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -242,3 +304,53 @@ def process_main(self, item=None):
 
         if new_tag:
             repository.associate_single_unit(self.parent.repo.repo_obj, new_tag)
+
+
+class AddBlobs(PluginStep):
+    """
+    Add Manifest and Blobs extracted in the ProcessManifest Step
+    """
+    def initialize(self):
+        """
+        Extract the tarfile to get all the layers from it.
+        """
+        # Brute force, extract the tar file for now
+        with contextlib.closing(tarfile.open(self.parent.file_path)) as archive:
+            archive.extractall(self.get_working_dir())
+
+    def get_iterator(self):
+        """
+        Return an iterator that will traverse the list of Units
+        that were present in the uploaded tarball.
+
+        :return: An iterable containing the Blobs and Manifests present in the uploaded tarball.
+        :rtype:  iterator
+        """
+        return iter(self.parent.v2_step_get_local_units.units_to_download)
+
+    def process_main(self, item=None):
+        """
+        Save blobs and manifest to repository
+
+        :param item: A Docker manifest or blob unit
+        :type      : pulp_docker.plugins.models.Blob or pulp_docker.plugins.models.Manifest
+        :return:     None
+        """
+        if isinstance(item, models.Blob):
```

Comment:
I have already extracted the manifest file before, therefore I am only extracting the blobs here.